src/Compilers/CSharp/Portable/Symbols/Source/SourceDelegateMethodSymbol.cs

```diff
@@ -42,15 +42,16 @@ protected void InitializeParameters(ImmutableArray<ParameterSymbol> parameters)
             DelegateDeclarationSyntax syntax,
             DiagnosticBag diagnostics)
         {
+            var compilation = delegateType.DeclaringCompilation;
             Binder binder = delegateType.GetBinder(syntax.ParameterList);
             RefKind refKind;
             TypeSyntax returnTypeSyntax = syntax.ReturnType.SkipRef(out refKind);
             var returnType = binder.BindType(returnTypeSyntax, diagnostics);
 
             // reuse types to avoid reporting duplicate errors if missing:
-            var voidType = binder.GetSpecialType(SpecialType.System_Void, diagnostics, syntax);
-            var objectType = binder.GetSpecialType(SpecialType.System_Object, diagnostics, syntax);
-            var intPtrType = binder.GetSpecialType(SpecialType.System_IntPtr, diagnostics, syntax);
+            var voidType = TypeSymbolWithAnnotations.Create(compilation, binder.GetSpecialType(SpecialType.System_Void, diagnostics, syntax));
+            var objectType = TypeSymbolWithAnnotations.Create(compilation, binder.GetSpecialType(SpecialType.System_Object, diagnostics, syntax));
```

Comment:
>var objectType = TypeSymbolWithAnnotations.Create(compilation, binder.GetSpecialType(SpecialType.System_Object, diagnostics, syntax)); [](start = 12, length = 134)

I think, depending on usage, we should treat this type as nullable when we are using the feature. A PROTOTYPE comment should be sufficient for now.